packages/combobox/src/Combobox.ts

```diff
@@ -83,6 +84,20 @@ export class Combobox extends Textfield {
     @property({ type: String, attribute: 'pending-label' })
     public pendingLabel = 'Pending';
 
+    protected pendingStateController: PendingStateController<this>;
+
+    constructor() {
+        super();
+        this.pendingStateController = new PendingStateController(this, {
+            pending: () => this.pending,
+            onPendingChange: (isPending: boolean) => {
+                if (isPending) {
+                    this.open = false;
+                }
+            },
+        });
+    }
+
```

Comment:
Should we not update the component's pending state here as well?

Reply:
No, we don't have to update component's state here because it will be called only when component's pending state has been changed.
The onPendingChange callback in the PendingStateController is called when the state of host(combobox) changes.